### PR TITLE
refactor(sheets-ui): cell-editor & formula-bar-editor

### PR DESCRIPTION
**Pull request by weird94**
close https://github.com/dream-num/univer-pro/issues/703
close https://github.com/dream-num/univer-pro/issues/1056
close https://github.com/dream-num/univer-pro/issues/2533
close https://github.com/dream-num/univer-pro/issues/2612
close https://github.com/dream-num/univer-pro/issues/2627
close https://github.com/dream-num/univer-pro/issues/2649
close https://github.com/dream-num/univer-pro/issues/2733

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Comment by zhaolixin7** [comment]
@weird94 编辑态时修改行高、列宽光标会自动跳到最前面

https://github.com/user-attachments/assets/fe739d3b-a235-41f6-a992-666e6eaa2461

File: packages/docs/src/commands/mutations/core-editing.mutation.ts [hunk 1]
```diff
@@ -34,6 +34,8 @@ export interface IRichTextEditingMutationParams extends IMutationCommonParams {
     // Do you need to compose the undo and redo of history, and compose of the change states.
     debounce?: boolean;
     options?: { [key: string]: boolean };
+    // Whether this mutation is from a sync operation.
```
Comment (Jocs) [inline packages/docs/src/commands/mutations/core-editing.mutation.ts:37]: What's sync operation?

Comment (weird94) [reply packages/docs/src/commands/mutations/core-editing.mutation.ts:37]: Mark this mutation is synced from other editor, like cell-editor to formula-bar-editor.

Comment (weird94) [reply packages/docs/src/commands/mutations/core-editing.mutation.ts:37]: > Mark this mutation is synced from other editor, like cell-editor to formula-bar-editor.

Mark this mutation is synced from other editor, like cell-editor to formula-bar-editor.

**Pull request merged**